zk.js/tests/system-programs/verifier_tests.ts

```diff
@@ -399,6 +407,14 @@ describe("Verifier Zero and One Tests", () => {
     }
   });
 
+  it("Wrong senderSpl", async () => {
+    for (const tx in transactions) {
+      const tmp_tx: Transaction = _.cloneDeep(transactions[tx]);
+      tmp_tx.params.accounts.senderSpl = SolanaKeypair.generate().publicKey;
```

Comment:
```suggestion
      const keypair = SolanaKeypair.generate();
      await airdropSol({
        connection: provider.connection,
        lamports: 10e9,
        recipientPublicKey: keypair.publicKey,
      });
      const tokenAccount = await getOrCreateAssociatedTokenAccount(
        provider.connection,
        keypair,
        MINT,
        keypair.publicKey,
      );
      await token.approve(
        provider.connection,
        keypair,
        tokenAccount.address,
        Transaction.getSignerAuthorityPda(
          merkleTreeProgramId,
          new PublicKey(
            VERIFIER_IDLS[tx].constants[0].value.slice(1, -1),
          ),
        ), //delegate
        keypair.publicKey, // owner
        shieldAmount * 10,
        [keypair],
      );
      tmp_tx.params.accounts.senderSpl = tokenAccount.address;
```

Reply:
Proposing the change the test so that it resembles the attack we are preventing with the error.
Just passing a random Solana public key will fail for a different reason for sure as well.

Reply:
Done, thanks!